### PR TITLE
[ADD] [database_cleanup] migrate to 11.0

### PR DESCRIPTION
**Pull request by hbrunn**

### CODE REVIEW

**Comment by hbrunn** [comment]
this will however duplicate all of transbot's commits. Should we add an instruction to squash those commits to the migration guide?

**Comment by StefanRijnhart** [comment]
These transbot commits are partial right? So they only include the changes to the migrated module? In that case it would be a good idea to include that instruction. Or maybe there is some obscure git command to do it automatically ;-)

File: database_cleanup/models/purge_columns.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            if model.model not in self.env:
+                continue
+            model_pool = self.env[model.model]
+            if not model_pool._auto:
+                continue
+            table2model.setdefault(
+                model_pool._table, (model.id, [])
+            )[1].append(model_pool)
+
+        for table, model_spec in table2model.iteritems():
```
Comment (StefanRijnhart) [inline database_cleanup/models/purge_columns.py]: Remaining instance of iteritems

Comment (Garamotte) [reply database_cleanup/models/purge_columns.py]: 2to3 normally takes care of this kind of modification, did you use it ?

**Comment by StefanRijnhart** [comment]
Interesting new use case: in 11.0/base, the usage field from the inherited table ir_actions is uninherited and added to some of the child tables. Can you make purge_columns so that it removes the column from the parent table first (which removes the column from the inheriting tables as well), otherwise an error occurs when purging all columns.

```
  File "/home/bruno/buildout/11.0/parts/server-tools/database_cleanup/models/purge_columns.py", line 51, in purge
    IdentifierAdapter(line.name)
  File "/home/bruno/buildout/11.0/parts/odoo/odoo/sql_db.py", line 155, in wrapper
    return f(self, *args, **kwargs)
  File "/home/bruno/buildout/11.0/parts/odoo/odoo/sql_db.py", line 232, in execute
    res = self._obj.execute(query, params)
psycopg2.ProgrammingError: cannot drop inherited column "openupgrade_legacy_11_0_usage"
```

File: database_cleanup/models/purge_tables.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    _description = 'Purge tables'
+
+    @api.model
+    def find(self):
+        """
+        Search for tables that cannot be instantiated.
+        Ignore views for now.
+        """
+        # Start out with known tables with no model
+        known_tables = ['wkf_witm_trans']
```
Comment (StefanRijnhart) [inline database_cleanup/models/purge_tables.py]: With the removal of the workflows, this table can and should now be purged as well.

**Review by StefanRijnhart — CHANGES_REQUESTED** [review]
Thanks! See inline comments for requested changes.

**Comment by hbrunn** [comment]
@StefanRijnhart thanks for the review, points taken. https://github.com/OCA/server-tools/pull/1009#issuecomment-334430872 is a feature request in my book that shouldn't be part of the migration PR

File: database_cleanup/models/purge_columns.py (hunk 1 above)
Comment (hbrunn) [reply database_cleanup/models/purge_columns.py]: no

**Review by StefanRijnhart — APPROVED** [review]
Fair enough! Thanks for the changes done.

**Comment by pedrobaeza** [comment]
@hbrunn maybe you can take the occasion for squashing some of the same author commits that are not too relevant?

**Review by DelfosCloud — APPROVED** [review]

**Comment by hbrunn** [comment]
@pedrobaeza done

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**